src/hotspot/share/opto/loopnode.cpp

```diff
@@ -3937,6 +3937,36 @@ bool PhaseIdealLoop::is_deleteable_safept(Node* sfpt) {
 
 //---------------------------replace_parallel_iv-------------------------------
 // Replace parallel induction variable (parallel to trip counter)
+// This optimization looks for patterns similar to:
+//
+//    int a = init2;
+//    for (int phi = init; phi < limit; phi += stride_con) {
+//      a += stride_con2;
+//    }
+//
+// and transforms it to:
+//
+//    int a = init2
+//    for (int phi = init; phi < limit; phi += stride_con) {
+//      a = init2 + (phi - init) * (stride_con2 / stride_con)
```

Comment:
A nit but in this pseudo code, shouldn't it be the post-incremented iv (i.e. `phi + stride_con`)? Because at that point, `phi` is not incremented, yet. You eventually replace the "parallel iv" (`phi2` in the actual code) which in the graph is the pre-incremented iv. Maybe naming the iv `phi` is a little bit confusing. 

You could also extend this to make it more explicit:
```
int iv2 = init2
int iv = init
loop:
   if (phi >= limit) goto exit
   phi += stride_con
   iv2 = init2 + (iv - init) * (stride_con2 / stride_con)
   goto loop
exit:
...
```

Reply:
Good catch! I found it quite difficult to write pseudo code expressing the IR structure after transformation. I've updated the comment (mostly taking your example). Thanks!